src/c/profile/discovery/discovery.c

```diff
@@ -130,9 +130,8 @@ bool read_info_headers(ucdrBuffer* ub)
     return valid;
```

Comment:
Maybe you could change the previous if by `return uxr_read_submessage_header(...);`

Reply:
You are right. Done!